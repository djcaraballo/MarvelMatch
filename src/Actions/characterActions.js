export const addCharacters = (characterArray) => {
  return {
    type: 'ADD_CHARACTERS',
    characters: characterArray
  }
}