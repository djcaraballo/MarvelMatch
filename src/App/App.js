import React, { Component } from 'react';
import './App.css';
import * as Cleaner from '../Utils/Cleaners/'
import NewHeroSignupForm from '../Components/NewHeroSignupForm/NewHeroSignupForm'
import QuizInstructions from '../Components/QuizInstructions/QuizInstructions'
import LandingPage from '../Components/LandingPage/LandingPage'

class App extends Component {

  async componentDidMount() {
    const cleanData = await Cleaner.getCharacterData()
    console.log(cleanData)
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          {/*<NewHeroSignupForm />*/}
          {/*<QuizInstructions />*/}
          <LandingPage />
        </header>
      </div>
    );
  }
}

export default App;
