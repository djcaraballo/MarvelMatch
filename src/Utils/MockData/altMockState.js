export const altMockState = {
      counter: 0,
      questionId: 1,
      question: "What color are your eyes?",
      answerOptions: [
        {
          type: "blue",
          content: "Blue"
        },
        {
          type: "brown",
          content: "Brown"
        },
        {
          type: "green",
          content: "Green"
        },
        {
          type: "other",
          content: "Other"
        },
      ],
      answer: '',
      answersCount: {
        blue: 1,
        brown: 0,
        green: 0,
        other: 0,
        blond: 0,
        black: 1,
        red: 0,
        auburn: 0,
        alternate: 0,
        human: 0,
        alien: 1,
        spirit: 0,
        mutant: 0,
        cyborg: 0
      },
      result: ''
    }

