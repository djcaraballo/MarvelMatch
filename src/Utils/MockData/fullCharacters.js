export const fullCharacters = [
  {
    name: "Captain America",
    id: 1009220,
    description: "Vowing to serve his country any way he could, young Steve Rogers took the super soldier serum to become America's one-man army. Fighting for the red, white and blue for over 60 years, Captain America is the living, breathing symbol of freedom and liberty.",
    siteURL: "http://marvel.com/characters/8/captain_america?utm_campaign=apiRef&utm_source=a4d95911b60999871aaed8a8c43823f6",
    appearance:{
      gender: "Male",
      race: "Human",
      height:["6'2","188 cm"],
      weight:["240 lb","108 kg"],
      eyeColor: "blue",
      hairColor: "blond"
    },
    powerstats: {
      intelligence:69,
      strength:19,
      speed:38,
      durability:55,
      power:60,
      combat:100
    },
    imageURL:"https://cdn.rawgit.com/akabab/superhero-api/0.2.0/api/images/sm/149-captain-america.jpg",
    matchTally:0
  },
  {
    name: "Daredevil",
    id: 1009262,
    description: "Abandoned by his mother, Matt Murdock was raised by his father, boxer \\\"Battling Jack\\\" Murdock, in Hell's Kitchen. Realizing that rules were needed to prevent people from behaving badly, young Matt decided to study law; however, when he saved a man from an oncoming truck, it spilled a radioactive cargo that rendered Matt blind while enhancing his remaining senses. Under the harsh tutelage of blind martial arts master Stick, Matt mastered his heightened senses and became a formidable fighter.",
    siteURL: "http://marvel.com/characters/11/daredevil?utm_campaign=apiRef&utm_source=a4d95911b60999871aaed8a8c43823f6",
    appearance:{
      gender: "Male",
      race: "Human",
      height:["6'0","183 cm"],
      weight:["200 lb","90 kg"],
      eyeColor: "blue",
      hairColor: "red"
    },
    powerstats: {
      intelligence:75,
      strength:13,
      speed:25,
      durability:35,
      power:61,
      combat:100
    },
    imageURL:"https://cdn.rawgit.com/akabab/superhero-api/0.2.0/api/images/sm/201-daredevil.jpg",
    matchTally:0
  },
]

export const uncleanAltCharacters = [
  {
    "id": 1,
    "name": "A-Bomb",
    "slug": "1-a-bomb",
    "powerstats": {
      "intelligence": 38,
      "strength": 100,
      "speed": 17,
      "durability": 80,
      "power": 24,
      "combat": 64
    },
    "appearance": {
      "gender": "Male",
      "race": "Human",
      "height": [
        "6'8",
        "203 cm"
      ],
      "weight": [
        "980 lb",
        "441 kg"
      ],
      "eyeColor": "Yellow",
      "hairColor": "No Hair"
    },
    "biography": {
      "fullName": "Richard Milhouse Jones",
      "alterEgos": "No alter egos found.",
      "aliases": [
        "Rick Jones"
      ],
      "placeOfBirth": "Scarsdale, Arizona",
      "firstAppearance": "Hulk Vol 2 #2 (April, 2008) (as A-Bomb)",
      "publisher": "Marvel Comics",
      "alignment": "good"
    },
    "work": {
      "occupation": "Musician, adventurer, author; formerly talk show host",
      "base": "-"
    },
    "connections": {
      "groupAffiliation": "Hulk Family; Excelsior (sponsor), Avengers (honorary member); formerly partner of the Hulk, Captain America and Captain Marvel; Teen Brigade; ally of Rom",
      "relatives": "Marlo Chandler-Jones (wife); Polly (aunt); Mrs. Chandler (mother-in-law); Keith Chandler, Ray Chandler, three unidentified others (brothers-in-law); unidentified father (deceased); Jackie Shorr (alleged mother; unconfirmed)"
    },
    "images": {
      "xs": "https://cdn.rawgit.com/akabab/superhero-api/0.2.0/api/images/xs/1-a-bomb.jpg",
      "sm": "https://cdn.rawgit.com/akabab/superhero-api/0.2.0/api/images/sm/1-a-bomb.jpg",
      "md": "https://cdn.rawgit.com/akabab/superhero-api/0.2.0/api/images/md/1-a-bomb.jpg",
      "lg": "https://cdn.rawgit.com/akabab/superhero-api/0.2.0/api/images/lg/1-a-bomb.jpg"
    }
  },
  {
    "id": 2,
    "name": "Abe Sapien",
    "slug": "2-abe-sapien",
    "powerstats": {
      "intelligence": 88,
      "strength": 28,
      "speed": 35,
      "durability": 65,
      "power": 100,
      "combat": 85
    },
    "appearance": {
      "gender": "Male",
      "race": "Icthyo Sapien",
      "height": [
        "6'3",
        "191 cm"
      ],
      "weight": [
        "145 lb",
        "65 kg"
      ],
      "eyeColor": "Blue",
      "hairColor": "No Hair"
    },
    "biography": {
      "fullName": "Abraham Sapien",
      "alterEgos": "No alter egos found.",
      "aliases": [
        "Langdon Everett Caul",
        "Abraham Sapien",
        "Langdon Caul"
      ],
      "placeOfBirth": "-",
      "firstAppearance": "Hellboy: Seed of Destruction (1993)",
      "publisher": "Dark Horse Comics",
      "alignment": "good"
    },
    "work": {
      "occupation": "Paranormal Investigator",
      "base": "-"
    },
    "connections": {
      "groupAffiliation": "Bureau for Paranormal Research and Defense",
      "relatives": "Edith Howard (wife, deceased)"
    },
    "images": {
      "xs": "https://cdn.rawgit.com/akabab/superhero-api/0.2.0/api/images/xs/2-abe-sapien.jpg",
      "sm": "https://cdn.rawgit.com/akabab/superhero-api/0.2.0/api/images/sm/2-abe-sapien.jpg",
      "md": "https://cdn.rawgit.com/akabab/superhero-api/0.2.0/api/images/md/2-abe-sapien.jpg",
      "lg": "https://cdn.rawgit.com/akabab/superhero-api/0.2.0/api/images/lg/2-abe-sapien.jpg"
    }
  },
]
