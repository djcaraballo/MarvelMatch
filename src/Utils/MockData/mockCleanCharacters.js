export const cleanCharacters = [
  {
    name: "A-Bomb (HAS)", 
    id: 1017100, 
    description: "Rick Jones has been Hulk's best bud since day one,…ses it like a giant bowling ball of destruction! ", 
    imageURL: "http://i.annihil.us/u/prod/marvel/i/mg/3/20/5232158de5b16/standard_fantastic.jpg", 
    siteURL: "http://marvel.com/characters/76/a-bomb?utm_campaig…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6"
  }
  {
    name: "A.I.M.", 
    id: 1009144, 
    description: "AIM is a terrorist organization bent on destroying the world.", 
    imageURL: "http://i.annihil.us/u/prod/marvel/i/mg/6/20/52602f21f29ec/standard_fantastic.jpg", 
    siteURL: "http://marvel.com/characters/77/aim.?utm_campaign=apiRef&utm_source=a4d95911b60999871aaed8a8c43823f6"
  }
  {
    name: "Abomination (Emil Blonsky)", 
    id: 1009146, 
    description: "Formerly known as Emil Blonsky, a spy of Soviet Yu…ransformed Bruce Banner into the incredible Hulk.", 
    imageURL: "http://i.annihil.us/u/prod/marvel/i/mg/9/50/4ce18691cbf04/standard_fantastic.jpg", 
    siteURL: "http://marvel.com/characters/81/abomination?utm_ca…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6"
  }
  {
    name: "Adam Warlock", 
    id: 1010354, description: "Adam Warlock is an artificially created human who …ocoon at a scientific complex called The Beehive.", 
    imageURL: "http://i.annihil.us/u/prod/marvel/i/mg/a/f0/5202887448860/standard_fantastic.jpg", 
    siteURL: "http://marvel.com/comics/characters/1010354/adam_w…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6"
  }
  {
    name: "Agent X (Nijo)", 
    id: 1011031, 
    description: "Originally a partner of the mind-altering assassin…inds crime family, which included Nijo's brother.", 
    imageURL: "http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/standard_fantastic.jpg", 
    siteURL: "http://marvel.com/characters/101/agent_x?utm_campa…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6"
  }
  {
    name: "Alex Wilder", 
    id: 1010755, 
    description: "Despite being the only one of the Runaways without…ect, as well as prodigy-level logic and strategy.", 
    imageURL: "http://i.annihil.us/u/prod/marvel/i/mg/2/c0/4c00377144d5a/standard_fantastic.jpg", 
    siteURL: "http://marvel.com/characters/2820/alex_wilder?utm_…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6"
  }
  {
    name: "Amun", 
    id: 1010905, 
    description: "Amun is a ruthless teenage assassin, employed by t…the ritual to initiate the Wasp's new chosen one.", 
    imageURL: "http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/standard_fantastic.jpg", 
    siteURL: "http://marvel.com/characters/140/amun?utm_campaign…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6"
  }
  {
    name: "Arnim Zola", 
    id: 1009740, 
    description: "The frail, dwarfish Arnim Zola was born in 1930s S… world's leading biochemist and genetic engineer.", 
    imageURL: "http://i.annihil.us/u/prod/marvel/i/mg/8/b0/4c00393a4cb7c/standard_fantastic.jpg", 
    siteURL: "http://marvel.com/characters/2790/arnim_zola?utm_c…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6"
  }
  {
    name: "Avengers", 
    id: 1009165, 
    description: "Earth's Mightiest Heroes joined forces to take on …s have come to be regarded as Earth's No. 1 team.", 
    imageURL: "http://i.annihil.us/u/prod/marvel/i/mg/9/20/5102c774ebae7/standard_fantastic.jpg", 
    siteURL: "http://marvel.com/characters/68/avengers?utm_campa…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6"
  }
  {
    name: "Azazel (Mutant)", 
    id: 1011766, 
    description: "A mutant from biblical times, Azazel is the ruler …at the Earth and everything on it belongs to him.", 
    imageURL: "http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/standard_fantastic.jpg", 
    siteURL: "http://marvel.com/characters/227/azazel?utm_campai…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6"
  }
]
