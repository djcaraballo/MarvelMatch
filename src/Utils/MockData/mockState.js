export const mockState = {
      counter: 0,
      questionId: 1,
      question: "What color are your eyes?",
      answerOptions: [
        {
          type: "blue",
          content: "Blue"
        },
        {
          type: "brown",
          content: "Brown"
        },
        {
          type: "green",
          content: "Green"
        },
        {
          type: "other",
          content: "Other"
        },
      ],
      answer: '',
      answersCount: {
        blue: 0,
        brown: 0,
        green: 0,
        other: 0,
        blond: 0,
        black: 0,
        red: 0,
        auburn: 0,
        alternate: 0,
        human: 0,
        alien: 0,
        spirit: 0,
        mutant: 0,
        cyborg: 0
      },
      result: ''
    }

