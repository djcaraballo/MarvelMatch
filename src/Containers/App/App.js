import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { addCharacters } from '../../Actions/characterActions';
import './App.css';
import Cleaner from '../../Utils/Cleaners'
import QuizInstructions from '../../Components/QuizInstructions/QuizInstructions'
import LandingPage from '../../Components/LandingPage/LandingPage'
import ErrorPage from '../../Components/ErrorPage/ErrorPage'
import Quiz from '../Quiz/Quiz'

const cleaner = new Cleaner()

export class App extends Component {
  async componentDidMount() {
    const characterData = await cleaner.combineCharacterObjects()
    this.props.addCharacters(characterData)
  }

  render() {
    return (
      <div className="App">
        <BrowserRouter>
          <Switch>
            <Route exact path='/quiz-land' component={QuizInstructions} />
            <Route exact path='/quiz' component={Quiz} />
            <Route path='/MarvelMatch' exact component={LandingPage} />
            <Route component={ErrorPage} />
          </Switch>
        </BrowserRouter>
        <footer>
          <p>"Data provided by Marvel. Â© 2014 Marvel"</p>
        </footer>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  characters: state.characters
})

export const mapDispatchToProps = (dispatch) => {
  return {
    addCharacters: characterData => {
      dispatch(addCharacters(characterData))
    }
  }
}

App.propTypes = {
  characters: PropTypes.array,
  addCharacters: PropTypes.func,
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
