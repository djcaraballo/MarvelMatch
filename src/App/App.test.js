import React from 'react';
import ReactDOM from 'react-dom';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { shallow } from 'enzyme'
import { cleanCharacters } from '../Utils/MockData/mockCleanCharacters'
import { addCharacters } from '../Actions/characterActions'

describe('App Component', () => {
  let wrapper
  let mockAddCharacters = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <App 
        characters={cleanCharacters}
        addCharacters={mockAddCharacters}
      />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps function', () => {
    it('should return an object with the characters', () => {
      const mockState = {
        characters: cleanCharacters
      }
      const expected = {
        characters: cleanCharacters
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps function', () => {
    it('should call dispatch with an addCharacters action when addCharacters is called', () => {
      const mockCharacter = cleanCharacters[0]
      const mockDispatch = jest.fn()
      const mockAction = addCharacters(mockCharacter)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addCharacters(mockCharacter)
      expect(mockDispatch).toHaveBeenCalledWith(mockAction)
    })
  })
})
