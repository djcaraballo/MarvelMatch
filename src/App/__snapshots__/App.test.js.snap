// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addCharacters={[MockFunction]}
    characters={
      Array [
        Object {
          "description": "Rick Jones has been Hulk's best bud since day one,…ses it like a giant bowling ball of destruction! ",
          "id": 1017100,
          "imageURL": "http://i.annihil.us/u/prod/marvel/i/mg/3/20/5232158de5b16/standard_fantastic.jpg",
          "name": "A-Bomb (HAS)",
          "siteURL": "http://marvel.com/characters/76/a-bomb?utm_campaig…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6",
        },
        Object {
          "description": "AIM is a terrorist organization bent on destroying the world.",
          "id": 1009144,
          "imageURL": "http://i.annihil.us/u/prod/marvel/i/mg/6/20/52602f21f29ec/standard_fantastic.jpg",
          "name": "A.I.M.",
          "siteURL": "http://marvel.com/characters/77/aim.?utm_campaign=apiRef&utm_source=a4d95911b60999871aaed8a8c43823f6",
        },
        Object {
          "description": "Formerly known as Emil Blonsky, a spy of Soviet Yu…ransformed Bruce Banner into the incredible Hulk.",
          "id": 1009146,
          "imageURL": "http://i.annihil.us/u/prod/marvel/i/mg/9/50/4ce18691cbf04/standard_fantastic.jpg",
          "name": "Abomination (Emil Blonsky)",
          "siteURL": "http://marvel.com/characters/81/abomination?utm_ca…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6",
        },
        Object {
          "description": "Adam Warlock is an artificially created human who …ocoon at a scientific complex called The Beehive.",
          "id": 1010354,
          "imageURL": "http://i.annihil.us/u/prod/marvel/i/mg/a/f0/5202887448860/standard_fantastic.jpg",
          "name": "Adam Warlock",
          "siteURL": "http://marvel.com/comics/characters/1010354/adam_w…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6",
        },
        Object {
          "description": "Originally a partner of the mind-altering assassin…inds crime family, which included Nijo's brother.",
          "id": 1011031,
          "imageURL": "http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/standard_fantastic.jpg",
          "name": "Agent X (Nijo)",
          "siteURL": "http://marvel.com/characters/101/agent_x?utm_campa…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6",
        },
        Object {
          "description": "Despite being the only one of the Runaways without…ect, as well as prodigy-level logic and strategy.",
          "id": 1010755,
          "imageURL": "http://i.annihil.us/u/prod/marvel/i/mg/2/c0/4c00377144d5a/standard_fantastic.jpg",
          "name": "Alex Wilder",
          "siteURL": "http://marvel.com/characters/2820/alex_wilder?utm_…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6",
        },
        Object {
          "description": "Amun is a ruthless teenage assassin, employed by t…the ritual to initiate the Wasp's new chosen one.",
          "id": 1010905,
          "imageURL": "http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/standard_fantastic.jpg",
          "name": "Amun",
          "siteURL": "http://marvel.com/characters/140/amun?utm_campaign…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6",
        },
        Object {
          "description": "The frail, dwarfish Arnim Zola was born in 1930s S… world's leading biochemist and genetic engineer.",
          "id": 1009740,
          "imageURL": "http://i.annihil.us/u/prod/marvel/i/mg/8/b0/4c00393a4cb7c/standard_fantastic.jpg",
          "name": "Arnim Zola",
          "siteURL": "http://marvel.com/characters/2790/arnim_zola?utm_c…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6",
        },
        Object {
          "description": "Earth's Mightiest Heroes joined forces to take on …s have come to be regarded as Earth's No. 1 team.",
          "id": 1009165,
          "imageURL": "http://i.annihil.us/u/prod/marvel/i/mg/9/20/5102c774ebae7/standard_fantastic.jpg",
          "name": "Avengers",
          "siteURL": "http://marvel.com/characters/68/avengers?utm_campa…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6",
        },
        Object {
          "description": "A mutant from biblical times, Azazel is the ruler …at the Earth and everything on it belongs to him.",
          "id": 1011766,
          "imageURL": "http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/standard_fantastic.jpg",
          "name": "Azazel (Mutant)",
          "siteURL": "http://marvel.com/characters/227/azazel?utm_campai…piRef&utm_source=a4d95911b60999871aaed8a8c43823f6",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <BrowserRouter>
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/signup"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/quiz-land"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/quiz"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
            />
          </Switch>
        </BrowserRouter>,
        <footer>
          <p>
            "Data provided by Marvel. © 2014 Marvel"
          </p>
        </footer>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/signup"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/quiz-land"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/quiz"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
            />
          </Switch>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/signup"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/quiz-land"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/quiz"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/signup",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/quiz-land",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/quiz",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            "Data provided by Marvel. © 2014 Marvel"
          </p>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "\\"Data provided by Marvel. © 2014 Marvel\\"",
          },
          "ref": null,
          "rendered": "\\"Data provided by Marvel. © 2014 Marvel\\"",
          "type": "p",
        },
        "type": "footer",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <BrowserRouter>
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/signup"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/quiz-land"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/quiz"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
              />
            </Switch>
          </BrowserRouter>,
          <footer>
            <p>
              "Data provided by Marvel. © 2014 Marvel"
            </p>
          </footer>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/signup"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/quiz-land"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/quiz"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
              />
            </Switch>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/signup"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/login"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/quiz-land"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/quiz"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />,
                <Route
                  component={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/signup",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/login",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/quiz-land",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/quiz",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              "Data provided by Marvel. © 2014 Marvel"
            </p>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "\\"Data provided by Marvel. © 2014 Marvel\\"",
            },
            "ref": null,
            "rendered": "\\"Data provided by Marvel. © 2014 Marvel\\"",
            "type": "p",
          },
          "type": "footer",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
